### DISCUSSION TITLE
Sever Property file generation seems to be incorrectly escaping special characters

### DISCUSSION THREAD

**Discussion by alexchomiak** [itzg/docker-minecraft-bedrock-server#593] [Q&A]
When configuring my docker-compose configuration with the following environment property:

`
LEVEL_TYPE: 'gardenofglass:gardenofglass.'
`

The generated server.properties file sets the level-type value as follows:
`
level-type=gardenofglass\:gardenofglass.
`

It seems to be escaping the semicolon, but not removing the forward slash /. 

This is causing world generation to behave incorrectly. Trying to manually correct the value just has this overwritten with the correct one. (even if omitting the environment variable)

**Comment by itzg** [comment]
That is standard java properties file behavior:

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Properties.html#store(java.io.Writer,java.lang.String)

Minecraft wiki also states that colon must be escaped:

https://minecraft.wiki/w/Server.properties#level-type

**Comment by itzg** [comment]
Are you sure that trailing period isn't the problem?

![image](https://github.com/user-attachments/assets/c66a2d32-e350-4435-9097-2bab488e9d94)
